File: lib/domain/backdoor/ports/backdoor_clear_database_output_port.rb
# frozen_string_literal: true

module Domain
  module Backdoor
    module Ports
      class BackdoorClearDatabaseOutputPort
        def on_success(success_dto)
          raise NotImplementedError, "Subclasses must implement on_success"
        end

        def on_failure(failure_dto)
          raise NotImplementedError, "Subclasses must implement on_failure"
        end
      end
    end
  end
end

File: crates/agrr-domain/src/backdoor/ports/backdoor_clear_database_output_port.rs
use crate::backdoor::dtos::{BackdoorClearDatabaseFailure, BackdoorClearDatabaseOutput};

/// Ruby: `Domain::Backdoor::Ports::BackdoorClearDatabaseOutputPort`
pub trait BackdoorClearDatabaseOutputPort {
    fn on_success(&mut self, success_dto: BackdoorClearDatabaseOutput);
    fn on_failure(&mut self, failure_dto: BackdoorClearDatabaseFailure);
}
